raft/src/main/java/org/apache/kafka/raft/ReplicatedCounter.java

```diff
@@ -87,6 +96,25 @@ public synchronized void handleCommit(BatchReader<Integer> reader) {
         }
     }
 
+    @Override
+    public synchronized void handleSnapshot(SnapshotReader<Integer> reader) {
+        try {
+            try (SnapshotReader<Integer> snapshot = reader) {
+                log.debug("Loading snapshot {}", snapshot.snapshotId());
+                for (List<Integer> batch : snapshot) {
+                    for (Integer value : batch) {
+                        log.debug("Setting value: {}", value);
+                        this.committed = value;
+                        this.uncommitted = value;
+                    }
+                }
+                log.debug("Finished loading snapshot. Set value: {}", this.committed);
+            }
+        } catch (IOException e) {
```

Comment:
Why do we need the second level here?

Reply:
I also changed a few places that were throwing `IOException`